var/spack/repos/builtin/packages/openblas/package.py

```diff
@@ -39,9 +39,14 @@ def install(self, spec, prefix):
             make_defs += ['BUILD_LAPACK_DEPRECATED=1']
 
         # Add support for OpenMP
-        # Note: Make sure your compiler supports OpenMP
         if '+openmp' in spec:
-            make_defs += ['USE_OPENMP=1']
+            # Note: Apple's Clang 7.3.0 does not support OpenMP.
+            # For now, raise an error if we compile with clang on OSX.
+            # TODO: one could still have a self-compiled clang-omp on OSX...
+            if (sys.platform == 'darwin') and (self.compiler.name == 'clang'):
```

Comment:
Do you want to pick up Apple's clang, or do you want to check whether the compiler supports OpenMP? The latter is more difficult, but the "right thing" to do. Using version numbers will always be iffy.

I would check the version number. Do you have an example of a broken version number? You could simply disallow versions `>=7`, with an appropriate comment, and open an issue with Spack to handle this in a cleaner way.